fabtests/unit/getinfo_test.c

```diff
@@ -126,8 +126,12 @@ static int validate_bit_combos(char *node, char *service, uint64_t flags,
 		return ret;
 	}
 
+	*info = NULL;
 	for (i = 0, fail = skipped = 0; i < cnt; i++) {
 		init(hints, combinations[i]);
+
+		if (*info)
+			fi_freeinfo(*info);
```

Comment:
To me, it makes the most sense to add this to the end of each loop rather than at the beginning. I think this way, there is still a leak on the last iteration of the combination but adding it at the end of the loop would solve that.

Reply:
This original function return the last successfully opened info object and the called will free it. This patch was mimicking the behavior.

Reply:
I agree with @aingerson -- it's much cleaner to call freeinfo after getinfo.  This likely means setting info=NULL, to avoid a double free later.

Reply:
sure. In new revision, I moved the ```fi_freeinfo()``` to end of loop, also set ```*info``` to ```NULL``` after free.